plugin/pkg/admission/broker/authsarcheck/admission.go

```diff
@@ -0,0 +1,143 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authsarcheck
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apiserver/pkg/admission"
+	kubeclientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	authorizationapi "k8s.io/client-go/pkg/apis/authorization/v1"
+
+	scadmission "github.com/kubernetes-incubator/service-catalog/pkg/apiserver/admission"
+)
+
+const (
+	pluginName = "BrokerAuthSarCheck"
+)
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register(pluginName, func(io.Reader) (admission.Interface, error) {
+		return NewSARCheck()
+	})
+}
+
+// sarcheck is an implementation of admission.Interface.
+// It enforces the creator of a broker has proper access to the auth credentials
+type sarcheck struct {
+	*admission.Handler
+	client kubeclientset.Interface
+}
+
+var _ = scadmission.WantsKubeClientSet(&sarcheck{})
+
+func convertToSARExtra(extra map[string][]string) map[string]authorizationapi.ExtraValue {
+	if extra == nil {
+		return nil
+	}
+
+	ret := map[string]authorizationapi.ExtraValue{}
+	for k, v := range extra {
+		ret[k] = authorizationapi.ExtraValue(v)
+	}
+
+	return ret
+}
+
+func (s *sarcheck) Admit(a admission.Attributes) error {
+	// need to wait for our caches to warm
+	if !s.WaitForReady() {
+		return admission.NewForbidden(a, fmt.Errorf("not yet ready to handle request"))
+	}
+	// only care about brokers
+	if a.GetResource().Group != servicecatalog.GroupName || a.GetResource().GroupResource() != servicecatalog.Resource("servicebrokers") {
+		return nil
+	}
+	serviceBroker, ok := a.GetObject().(*servicecatalog.ServiceBroker)
+	if !ok {
+		return errors.NewBadRequest("Resource was marked with kind ServiceBroker, but was unable to be converted")
+	}
+	glog.V(5).Infof("Retrieved serviceBroker %v", serviceBroker)
+
+	if serviceBroker.Spec.AuthInfo == nil {
+		// no auth secret to check
+		return nil
+	}
+
+	var secretRef *v1.ObjectReference
+	if serviceBroker.Spec.AuthInfo.Basic != nil {
+		secretRef = serviceBroker.Spec.AuthInfo.Basic.SecretRef
+	} else if serviceBroker.Spec.AuthInfo.Bearer != nil {
+		secretRef = serviceBroker.Spec.AuthInfo.Bearer.SecretRef
+	} else if serviceBroker.Spec.AuthInfo.BasicAuthSecret != nil {
+		// TODO: this field is deprecated, remove in v1beta1
+		secretRef = serviceBroker.Spec.AuthInfo.BasicAuthSecret
+	}
+	userInfo := a.GetUserInfo()
+
+	sar := &authorizationapi.SubjectAccessReview{
+		Spec: authorizationapi.SubjectAccessReviewSpec{
+			ResourceAttributes: &authorizationapi.ResourceAttributes{
+				Namespace: secretRef.Namespace,
+				Verb:      "get",
+				Group:     "", // core API does not have a group
+				Version:   "v1",
+				Resource:  "secrets",
```

Comment:
You can reference `apiv1.ResourceSecrets.String()` constant here for resource name if you like.
https://github.com/kubernetes-incubator/service-catalog/blob/master/vendor/k8s.io/client-go/pkg/api/v1/types.go#L4140